File: languages/java/secure-memory/src/test/java/com/godaddy/asherah/securememory/protectedmemoryimpl/ProtectedMemoryAllocationFailedTest.java
package com.godaddy.asherah.securememory.protectedmemoryimpl;

import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.*;

class ProtectedMemoryAllocationFailedTest {
  static String message = "Test Message";

  @Test
  void protectedMemoryAllocationFailedConstructorTest() {
    ProtectedMemoryAllocationFailed ex = new ProtectedMemoryAllocationFailed(message);
    assertEquals(message, ex.getMessage());
  }
}

File: languages/csharp/SecureMemory/SecureMemory.Tests/ProtectedMemoryImpl/ProtectedMemoryAllocationFailedTest.cs
using GoDaddy.Asherah.SecureMemory.ProtectedMemoryImpl;
using Xunit;

namespace GoDaddy.Asherah.SecureMemory.Tests.ProtectedMemoryImpl
{
    [Collection("Logger Fixture collection")]
    public class ProtectedMemoryAllocationFailedTest
    {
        private const string Message = "Failure message";

        [Fact]
        private void ProtectedMemoryAllocationFailedConstructorTest()
        {
            ProtectedMemoryAllocationFailedException exception = new ProtectedMemoryAllocationFailedException(Message);
            Assert.Equal(Message, exception.Message);
        }
    }
}
